Assets/Scripts/PlayerManager.cs

```diff
@@ -47,28 +48,46 @@ private void Start()
 
     private void Update()
     {
-        if (CanRest)
+        if (CanRest && Input.GetKeyDown(KeyCode.F))
         {
-            bool isResting = Input.GetKey(KeyCode.E);
-            UIManager.Instance.UpdateRestingText(isResting);
+            // Temporary Controls for Minigame
+            MinigameManager.Instance.LoadRandomMinigame();   
+        }
+    }
 
-            if (isResting)
-            {
-                HandleIncreaseStamina();
-            } 
-            else
-            {
-                // Temporary Controls for Minigame
-                if (Input.GetKeyDown(KeyCode.P))
-                {
-                    HandleTriggerStartMinigame();
-                }
-            }
+    private void FixedUpdate()
+    {
+        HandleDecreaseStamina();
+
+        float minDistance = distToClosestEnemy();
+
+        if (minDistance < dangerRadius)
```

Comment:
Nitpick, but I think you can do something like:

```
if ((float minDistance = distToClosestEnemy()) < dangerRadius)
```
to remove some lines of code

Reply:
C# won't let me do it like that - either way I have to instantiate minDistance before the "if" statement.